documentation/graph-analytics.adoc

```diff
@@ -1,3 +1,157 @@
+RETURN graphName
+----
+
+For a more detailed example including the use of query parameters for the Aura API credentials, updates to the in-memory graph, and write-back to AuraDB, see the link:https://neo4j.com/docs/graph-data-science/current/aura-graph-analytics/quickstart/[Aura Graph Analytics quickstart].
+
+For more information about the Graph Analytics Cypher API, see the link:https://neo4j.com/docs/graph-data-science/current/aura-graph-analytics/cypher[Graph Data Science documentation].
+
+== Graph Analytics with the Python client
+
+Compared to the Cypher API, the Python client adds the option to run algorithms on data from non-Neo4j data sources such as Pandas DataFrames.
+
+To use the Graph Analytics features in Neo4j Aura with Python, you need to install the `graphdatascience` package.
+
+[source, bash, copy=true]
+----
+pip install graphdatascience
+----
+
+The entry point for managing Aura Graph Analytics sessions is the `GdsSessions` class.
+
+[source, python, copy=true]
+----
+from graphdatascience.session import GdsSessions, AuraAPICredentials
+
+CLIENT_ID = "<Aura API Client ID>"
+CLIENT_SECRET = "<Aura API Client Secret>"
+PROJECT_ID = None
+# Create a new GdsSessions object
+sessions = GdsSessions(api_credentials=AuraAPICredentials(CLIENT_ID, CLIENT_SECRET, PROJECT_ID))
+----
+
+Use the `sessions` object to create an Aura Graph Analytics session.
+
+In the following example, the session is attached to an AuraDB instance.
+Here the username and password are the credentials of your AuraDB instance, _not_ your Aura API credentials.
+
+[source, python, copy=true]
+----
+from graphdatascience.session import DbmsConnectionInfo, SessionMemory
+
+db_connection = DbmsConnectionInfo(
+     uri="<AuraDB instance URI>",
+     username="<AuraDB username>",
+     password="<AuraDB password>"
+)
+
+gds = sessions.get_or_create(
+    session_name="my-session",
+    memory=SessionMemory.m_4GB,
+    db_connection=db_connection,
+)
+----
+
+With the session ready, add some data to the AuraDB instance with the `run_cypher` method.
+
+[source, python, copy=true]
+----
+gds.run_cypher("""
+    CREATE
+      (a:User {name: 'Alice'}),
+      (b:User {name: 'Bob'}),
+      (c:User {name: 'Charlie'}),
+      (d:User {name: 'Dan'}),
+      (e:User {name: 'Eve'}),
+      (f:User {name: 'Frank'}),
+      (a)-[:LINK]->(b),
+      (a)-[:LINK]->(c),
+      (c)-[:LINK]->(e),
+      (e)-[:LINK]->(d),
+      (e)-[:LINK]->(f)
+""")
+----
+
+Then, create the remote graph `G` and run PageRank on it.
+
+[source, python, copy=true]
+----
+G, result = gds.graph.project(
+    graph_name="my-graph",
+    query="""
+        CALL {
+          MATCH (n)
+          OPTIONAL MATCH (n)-->(m)
+          RETURN n, m
+        }
+        RETURN gds.graph.project.remote(n, m, {})
+    """,
+)
+
+gds.pageRank.stream(G)
+----
+
+Finally, delete the session when you no longer need it.
+
+[source, python, copy=true]
+----
+sessions.delete(session_name="my-session")
+----
```

Comment:
This could also be `gds.delete()`

Reply:
Yes, although I thought we explicitly described and used the `sessions` object so it may be worth using it again. The client docs make it clear that you can do both.
What would be the preferred option from your side?

Reply:
I prefer the shorter one.

Reply:
Sounds good. I'll rename `gds` to `my_session` to make it clearer.